pmlc/dialect/comp/ir/types.h

```diff
@@ -63,6 +63,47 @@ class RuntimeType : public mlir::Type {
   }
 };
 
+// ============================================================================
+// DeviceType;
+// ============================================================================
+class ExecEnvType;
+
+struct DeviceStorage : public RuntimeTypeStorage {
+  explicit DeviceStorage(ExecEnvRuntime runtime)
+      : RuntimeTypeStorage(runtime) {}
+
+  using KeyTy = ExecEnvRuntime;
```

Comment:
I don't understand this. Are we saying that the Device and the ExecEnvRuntime are 1-to-1?

Reply:
As discussed -- removed runtime from device type until/unless we actually need it.